tests/02_eradiate/01_unit/scenes/atmosphere/test_heterogeneous.py

```diff
@@ -306,11 +313,92 @@ def test_heterogeneous_blend_switches(mode_mono):
         particle_layers=[ParticleLayer()],
     )
 
-    # Purely absorbing atmosphere + particle layer combination is not allowed
-    with pytest.raises(ValueError):
-        HeterogeneousAtmosphere(
-            molecular_atmosphere=MolecularAtmosphere.ussa_1976(
-                has_absorption=True, has_scattering=False
+
+def test_heterogeneous_absorbing_mol_atm(
+    mode_ckd, tmp_path, absorbing_only, scattering_only
+):
+    """
+    Phase function weights are correct when the molecular atmosphere is
+    absorbing-only and the particle layer is either absorbing-only or
+    scattering-only.
+    """
+    # Create the heterogeneous atmosphere
+    pl_bottom = 1.0 * ureg.km  # arbitrary
+    pl_top = 4.0 * ureg.km  # arbitrary
+    particle_layer = ParticleLayer(
+        bottom=pl_bottom,
+        top=pl_top,
+        dataset=scattering_only,
+    )
+    atmosphere1 = HeterogeneousAtmosphere(
+        molecular_atmosphere=MolecularAtmosphere.afgl_1986(
+            has_absorption=True,
+            has_scattering=False,
+        ),
+        particle_layers=particle_layer,
+        geometry={"type": "spherical_shell"},
+    )
+    atmosphere2 = attrs.evolve(
+        atmosphere1,
+        particle_layers=ParticleLayer(
+            bottom=pl_bottom,
+            top=pl_top,
+            dataset=absorbing_only,
+        ),
+    )
```

Comment:
Rather than instantiating two objects and repeating a lot of code, making this two iterations of a parametrised tests would be clearer.